charms/istio-pilot/src/istioctl.py

```diff
@@ -11,34 +12,50 @@ class IstioctlError(Exception):
 
 class Istioctl:
     def __init__(
-        self, istioctl_path: str, namespace: str = "istio-system", profile: str = "minimal"
+        self,
+        istioctl_path: str,
+        namespace: str = "istio-system",
+        profile: str = "minimal",
+        istioctl_extra_flags: Optional[List[str]] = [],
```

Comment:
Initialising this to `[]` has the unexpected side effect of sharing the default value between all instances of the class ([ref 1](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments), [ref 2](https://stackoverflow.com/questions/4841782/python-constructor-and-default-value)).  Can we change this to `= None` and then have an `if istioctl_extra_flags = None` in the init?
 an unexpected side effect because `[]` is mutable.  It basically means all instances of `Istioctl` will share

Reply:
Yeah, I can definitely do that.